lib/strarray.c

```diff
@@ -371,15 +371,8 @@ EXPORTED strarray_t *strarray_splitm(char *buf, const char *sep, int flags)
             *q = '\0';
         }
         if (*p) {
-            if (flags & STRARRAY_UNIQ) {
-                int found = -1;
-                if ((flags & STRARRAY_LCASE))
-                    found = strarray_find_case(sa, p, 0);
-                else
-                    found = strarray_find(sa, p, 0);
-
-                if (found != -1) continue;
-            }
+            if ((flags & STRARRAY_UNIQ) && strarray_find(sa, p, 0) != -1)
```

Comment:
nit: it did not occur to me that splitm with UNIQ is O(n^2) in disguise. But I guess since this hasn't shown to be a problem so far, it is fine.

Reply:
I would like to have a `strset` type (probably built on `hash_table`) and convert all the existing code that uses strarrays as a set to use that instead.  But that use case clearly hasn't been a problem so far (same algorithm as `strarray_intersect()`), so I've consciously just gone with the simplest thing for now, rather than waste time trying to bolt an awkward optimisation onto what is already the wrong data type.

If we find this is a problem in practice, that'll be a good prompt to do sets properly...

Reply:
Meh - 'n' is approximately exactly equal to one in most of these cases.